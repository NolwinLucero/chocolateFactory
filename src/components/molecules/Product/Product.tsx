import React from 'react';
import styled from 'styled-components';
import { ProductSchema } from '../../../schemas/product';

interface Props {
  product: ProductSchema;
  credits: number;
  OnBuyHandler: (e: React.FormEvent, id: number) => void;
}

export const Product: React.FC<Props> = ({
  product,
  credits,
  OnBuyHandler,
}) => {
  return (
    <>
      <StyledProductDiv>
        <div>{product.name}</div>
        <img
          alt={product.name}
          src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Gooey-Chocolate-Caramel-Bars_EXPS_CMZ18_10572_D10_26_5b-1.jpg?fit=200,200"
        ></img>
        <div>Stock: {product.stock}</div>
        <div>{product.price}</div>
        <button
          disabled={credits < parseFloat(product.price?.replace('$', ''))}
          data-testid={product.id.toString()}
          id={product.id.toString()}
          onClick={(e) => OnBuyHandler(e, product.id)}
        >
          Buy
        </button>
      </StyledProductDiv>
    </>
  );
};

const StyledProductDiv = styled.div`
  background-color: gold;
  padding-bottom 20px;
  padding-top: 20px;
`;
