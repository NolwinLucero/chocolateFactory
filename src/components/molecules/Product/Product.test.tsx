import React from 'react';
import ReactDOM from 'react-dom';
import { Product } from './Product';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

const credits = 5;
const product = {
  id: 1,
  name: 'Caramel',
  price: '$3.10',
  stock: 3,
};

const OnBuyHandler = () => {
  alert('clicked');
};

window.alert = jest.fn();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Product
      product={product}
      credits={credits}
      OnBuyHandler={OnBuyHandler}
    ></Product>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('add credits button', () => {
  const { getByTestId } = render(
    <Product
      product={product}
      credits={credits}
      OnBuyHandler={OnBuyHandler}
    ></Product>
  );

  const button = getByTestId('1');
  fireEvent.click(button);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Product
        product={product}
        credits={credits}
        OnBuyHandler={OnBuyHandler}
      ></Product>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
