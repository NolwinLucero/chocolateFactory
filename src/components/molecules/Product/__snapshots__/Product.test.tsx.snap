// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdvvtL bbdKiI"
>
  <div>
    Caramel
  </div>
  <img
    alt="Caramel"
    src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Gooey-Chocolate-Caramel-Bars_EXPS_CMZ18_10572_D10_26_5b-1.jpg?fit=200,200"
  />
  <div>
    Stock: 
    3
  </div>
  <div>
    $3.10
  </div>
  <button
    data-testid="1"
    disabled={false}
    id="1"
    onClick={[Function]}
  >
    Buy
  </button>
</div>
`;
